test/mocks/overrides/override_map_view_model.dart

```diff
@@ -49,6 +51,27 @@ class MockMapViewModel extends AutoDisposeAsyncNotifier<MapDetails>
 
   @override
   Completer<GoogleMapController> get mapController => _mapController;
+
+  @override
+  void disableFollowUser() {
+    _followUser = false;
+  }
+
+  @override
+  void enableFollowUser() {
+    _followUser = true;
+  }
+
+  @override
+  double get initialZoom => _initialZoomLevel;
+
+  @override
+  Future<void> moveCamera(LatLng target) async {
+    if (!_followUser) return;
+    final GoogleMapController controller = await _mapController.future;
+    controller
+        .animateCamera(CameraUpdate.newLatLngZoom(target, _initialZoomLevel));
+  }
```

Comment:
Good initiative to update the mock :+1: 
I just think that because this is a mock, we shouldn't perform the implementation inside it but rather pass a function as contructor parameter. Like that, we can create mocks with various behaviors.
Concretely, I believe we could have something like that:

Basically, I am just injecting the functions `_disableFollowUser`, `_enableFollowUser` and `_moveCamera` into the constructor.

```dart
class MockMapViewModel extends AutoDisposeAsyncNotifier<MapDetails>
    implements MapViewModel {
  final Future<MapDetails> Function() _build;
  final Future<void> Function() _onRefresh;
  final void Function(GoogleMapController) _onMapCreated;
  final Set<Circle> _circles;
  final Future<void> Function(LatLng) _redrawCircle;
  final Completer<GoogleMapController> _mapController;

  final void Function() _disableFollowUser;
  final void Function() _enableFollowUser;
  final Future<void> Function(LatLng) _moveCamera;

  final double _initialZoomLevel = 17.0;

  MockMapViewModel({
    Future<MapDetails> Function()? build,
    Future<void> Function()? onRefresh,
    Future<void> Function(LatLng)? animateCamera,
    void Function(GoogleMapController)? onMapCreated,
    Future<void> Function(LatLng)? redrawCircle,
    Set<Circle>? circles,
    Completer<GoogleMapController>? mapController,
    void Function()? disableFollowUser,
    void Function()? enableFollowUser,
    void Function(LatLng)? moveCamera,
  })  : _build = build ??
            (() async => throw Exception("Location services are disabled.")),
        _onRefresh = onRefresh ?? (() async {}),
        _onMapCreated = onMapCreated ?? ((_) {}),
        _circles = circles ?? {},
        _redrawCircle = redrawCircle ?? ((_) async {}),
        _mapController = mapController ?? Completer(),
        _disableFollowUser = disableFollowUser ?? (() {}),
        _enableFollowUser = enableFollowUser ?? (() {}),
        _moveCamera = animateCamera ?? ((_) async {});

  @override
  Future<MapDetails> build() => _build();

  @override
  Future<void> refresh() => _onRefresh();

  @override
  void onMapCreated(GoogleMapController controller) =>
      _onMapCreated(controller);

  @override
  Set<Circle> get circles => _circles;

  @override
  Future<void> redrawCircle(LatLng target) => _redrawCircle(target);

  @override
  Completer<GoogleMapController> get mapController => _mapController;

  @override
  void disableFollowUser() => _disableFollowUser();

  @override
  void enableFollowUser() => _enableFollowUser();

  @override
  double get initialZoom => _initialZoomLevel;

  @override
  Future<void> moveCamera(LatLng target) => _moveCamera(target);
}
```
What do you think ?

Note: If you go with making `_initialZoomLevel` public, I think that we can remove it entirely from the mock.

Reply:
Yes good catch! When I wrote this I was confused because it's actually a fake and not a mock.
Fixed in 19015afbc43456c531524c7b1b00417d7566eb25